### PR TITLE
Drop support for EOL Python 3.3

### LINKED ISSUE #192: End Explicit Support for Python 3.3
Python 3.3 has been EoL'ed since September, and more and more major packages are dropping support, making maintenance more difficult. Pytest, for example, [dropped support in version 3.3.0](https://github.com/pytest-dev/pytest/blob/master/CHANGELOG.rst#pytest-330-2017-11-23).

I think that unless there is a really good reason, we should stop explicitly supporting Python 3.3 going forward.

### PR DESCRIPTION
**Pull request by hugovk**
Fix #192.

Also remove some redundant code for Python < 2.7 and 3.0-3.2, and some minor tidy-ups.

### CODE REVIEW

**Comment by ankostis** [comment]
great!

File: pypiserver/bottle.py [hunk 1]
```diff
@@ -89,17 +89,15 @@ def _cli_patch(args):
 
             def json_dumps(data):
                 raise ImportError(
-                    "JSON support requires Python 2.6 or simplejson.")
+                    "JSON support requires simplejson.")
```
Comment (ankostis) [inline pypiserver/bottle.py]: I would just not touch the `bottle.py` - that's an "upstream-derived" file for *pypiserver*, we don't maintain it.

I know that your changes are not invasive and make sanse, but 
there is another reason to keep this file pristine: 
when we upgrade *bottle*, [sometimes](https://github.com/pypiserver/pypiserver/commit/2d3f7bf) certain changes are needed to fix things.  Leaving `bottle.py` untouched makes these cases stand out, and be easier to redo those changes later, on each new upgrade.

Comment (hugovk) [reply pypiserver/bottle.py]: Makes sense, I'll revert those.

**Comment by hugovk** [comment]
Rebased to resolve conflicts.

Note: latest master is also now failing: https://travis-ci.org/hugovk/pypiserver/builds/376047005

**Pull request merged**